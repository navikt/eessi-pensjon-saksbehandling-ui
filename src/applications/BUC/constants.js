export const BUC_MODE = 'BUC/MODE'
