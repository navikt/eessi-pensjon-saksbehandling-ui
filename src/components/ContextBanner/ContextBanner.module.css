.content {
  background-color: var(--a-surface-subtle);
  border-bottom: 1px solid var(--a-border-strong);
}

.tag {
  font-size: var(--a-font-size-small);
}
