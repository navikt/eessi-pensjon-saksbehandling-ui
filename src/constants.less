// Farger og fargepallett
// ------------------------------------
@white:       #ffffff;
@white-off:   #efefef;
@black:       #000000;
@purple:      #634689;
@purple-light:#826ba1;
@gray:        #d8d7d6;
@gray-dark:   #b7b1a9;
@orange:      #ff9100;
@orange-dark: #9b5901;
@blue-light:  #66cbec;
@blue:        #4990e2;
@blue-nav:    #0067c5;
@green:       #06893a;

// NAV farge
@navRod:      #C30000;
@navOransje:  #FF9100;
@navLimeGronn:#A2AD00;
@navGronn:    #06893A;
@navLilla:    #634689;
@navDypBla:   #005B82;
@navBla:      #0067C5;
@navLysBla:   #66CBEC;

//Tekst digitalt
@navMorkGra:  #3E3832;
@navGra80:    #59514B;
@navGra60:    #78706A;
@navGra40:    #B7B1A9;
@navGra20:    #C6C2BF;
@navLysGra:   #E9E7E7;

@pension-orange: rgb(255, 167, 51);
@pension-orange-alpha: rgba(255, 167, 51, 0.5);
@pension-orange-alpha-80: rgba(255, 167, 51, 0.8);
@high-contrast-orange: #ffc40d;
@form-dark-gray: #78706A;
@border-gray: #59514B;
@red-fail: #BA3A26;
// Typografi og fonter
// ------------------------------------

@font-color:  @black;
