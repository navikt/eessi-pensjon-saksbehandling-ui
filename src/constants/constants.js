export const SAKSBEHANDLER = 'SAKSBEHANDLER'
export const BRUKER = 'BRUKER'

export const P2000 = 'P2000'
export const P2100 = 'P2100'
export const P2200 = 'P2200'
export const P4000 = 'P4000'
export const P6000 = 'P6000'
export const PINFO = 'PINFO'

export const PINFO_FILE = 'PINFO.json'
