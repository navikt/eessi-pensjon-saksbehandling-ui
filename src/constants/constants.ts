export const SAKSBEHANDLER = 'SAKSBEHANDLER'

export const P2000 = 'P2000'
export const P2100 = 'P2100'
export const P2200 = 'P2200'
export const P4000 = 'P4000'
export const P6000 = 'P6000'

export const BRUKEROVERSIKT = 'brukeroversikt'
export const KRAVKONTEKST = 'kravkontekst'
export const VEDTAKSKONTEKST = 'vedtakskontekst'
