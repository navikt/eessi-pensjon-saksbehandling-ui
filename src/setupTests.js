import Enzyme, { shallow, render, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
// import { expect } from 'chai'
import cheerio from 'cheerio'

Enzyme.configure({ adapter: new Adapter() })

global.shallow = shallow
global.render = render
global.mount = mount

global.cheerio = cheerio
