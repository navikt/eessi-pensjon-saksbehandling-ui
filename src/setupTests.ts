import Enzyme, { shallow, render, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { act } from 'react-dom/test-utils'
Enzyme.configure({ adapter: new Adapter() });

(global as any).shallow = shallow;
(global as any).render = render;
(global as any).mount = mount;
(global as any).act = act

// eslint-disable-next-line no-undef
HTMLCanvasElement.prototype.getContext = jest.fn()
window.scrollTo = jest.fn()
window.location.reload = jest.fn()
window.fetch = require('jest-fetch-mock')
Object.defineProperty(window, 'getComputedStyle', {
  value: () => ({
    getPropertyValue: () => ('')
  })
})
