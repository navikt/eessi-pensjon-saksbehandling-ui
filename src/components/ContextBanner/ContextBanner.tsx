import { BUCMode } from 'applications/BUC'
import ExternalLink from 'assets/icons/line-version-logout'
import { HighContrastLink, HorizontalSeparatorDiv } from 'components/StyledComponents'
import { State } from 'declarations/reducers'
import { Person, PesysContext } from 'declarations/types'
import BUCWebSocket from 'applications/BUC/websocket/WebSocket'
import _ from 'lodash'
import { linkLogger, standardLogger } from 'metrics/loggers'
import { HoyreChevron } from 'nav-frontend-chevron'
import { EtikettLiten } from 'nav-frontend-typografi'
import { theme, themeKeys, themeHighContrast } from 'nav-styled-component-theme'
import React, { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useSelector } from 'react-redux'
import styled, { ThemeProvider } from 'styled-components'

const DivWithLinks = styled.div`
  padding: 0.5rem 2rem;
  display: flex;
  flex-direction: row-reverse;
`
const SeparatorSpan = styled.span`
  padding: 0rem 0.5rem
`
const Content = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  background-color: ${({ theme }) => theme.type === 'themeHighContrast' ? 'black' : 'lightgrey'};
`
const Context = styled.div`
 padding: 0.5rem 2rem;
 display: flex;
 align-items: center;
 * {
   font-size: ${({ theme }) => theme.type === 'themeHighContrast' ? '1.5rem' : 'inherit'};
   line-height: ${({ theme }) => theme.type === 'themeHighContrast' ? '1.5rem' : 'inherit'};
 }
`
const Tag = styled(EtikettLiten)`
 padding: 0rem 0.5rem;
 margin-left: 0.25rem;
 margin-right: 0.25rem;
`

interface ContextBannerProps {
  highContrast: boolean
  mode: BUCMode
}

interface ContextBannerSelector {
  person: Person | undefined
  pesysContext?: PesysContext
  sakType?: string
}

const mapState = (state: State): ContextBannerSelector => ({
  person: state.app.person,
  pesysContext: state.app.pesysContext,
  sakType: state.app.params.sakType
})

const ContextBanner: React.FC<ContextBannerProps> = ({
  highContrast, mode
}: ContextBannerProps): JSX.Element => {
  const _theme = highContrast ? themeHighContrast : theme
  const linkColor = _theme[themeKeys.MAIN_INTERACTIVE_COLOR]
  const [mounted, setMounted] = useState<boolean>(false)
  const { t } = useTranslation()
  const { person, pesysContext, sakType }: ContextBannerSelector =
    useSelector<State, ContextBannerSelector>(mapState)

  useEffect(() => {
    if (!mounted) {
      standardLogger('context', {
        pesys: pesysContext,
        sakType: sakType
      })
      setMounted(true)
    }
  }, [mounted, pesysContext, sakType])

  return (
    <ThemeProvider theme={highContrast ? themeHighContrast : theme}>
      <Content>
        <Context>
          <BUCWebSocket
            fnr={_.get(person, 'aktoer.ident.ident')}
            avdodfnr=''
          />
          <HoyreChevron />
          <Tag>
            <span>{t('ui:youComeFrom')}</span>
            <strong>{pesysContext}</strong>.
          </Tag>
          {sakType && (
            <Tag>
              <span>{t('buc:form-caseType')}: </span>
              <strong>{sakType}</strong>
            </Tag>
          )}
        </Context>
        <DivWithLinks>
          <HighContrastLink
            target='_blank'
            data-amplitude='links.rettskilder'
            href='https://lovdata.no/pro/#document/NAV/rundskriv/v2-45-03'
            onClick={(e: React.MouseEvent) => linkLogger(e, { mode: mode })}
          >
            {t('ui:lawsource')}
            <HorizontalSeparatorDiv data-size='0.5' />
            <ExternalLink color={linkColor} />
          </HighContrastLink>
          <SeparatorSpan>
            â€¢
          </SeparatorSpan>
          <HighContrastLink
            target='_blank'
            data-amplitude='links.hjelpe'
            href='https://navno.sharepoint.com/sites/fag-og-ytelser-regelverk-og-rutiner/SitePages/Pensjon-.aspx'
            onClick={(e: React.MouseEvent) => linkLogger(e, { mode: mode })}
          >
            {t('ui:help')}
            <HorizontalSeparatorDiv data-size='0.5' />
            <ExternalLink color={linkColor} />
          </HighContrastLink>
        </DivWithLinks>
      </Content>
    </ThemeProvider>
  )
}

export default ContextBanner
