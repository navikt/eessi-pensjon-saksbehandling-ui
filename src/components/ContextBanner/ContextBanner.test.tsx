import BucWebSocket from 'applications/BUC/websocket/WebSocket'
import { VEDTAKSKONTEKST } from 'constants/constants'
import { mount, ReactWrapper } from 'enzyme'
import { stageSelector } from 'setupTests'
import mockPerson from 'mocks/app/person'
import ContextBanner, { Context, ContextBannerProps, ContextBannerSelector, DivWithLinks, Tag } from './ContextBanner'

jest.mock('applications/BUC/websocket/WebSocket', () => ({ title, children }: any) => (
  <div title={title} className='websocket'>{children}</div>
))

const mockPesysContext = VEDTAKSKONTEKST
const mockSakType = 'Generell'

const defaultSelector: ContextBannerSelector = {
  gettingSakType: false,
  person: mockPerson,
  pesysContext: mockPesysContext,
  sakType: mockSakType
}

describe('components/ContextBanner/ContextBanner', () => {
  let wrapper: ReactWrapper
  const initialMockProps: ContextBannerProps = {
    mode: 'buclist'
  }

  beforeEach(() => {
    stageSelector(defaultSelector, {})
    wrapper = mount(<ContextBanner {...initialMockProps} />)
  })

  it('Render: match snapshot', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Render: has proper HTML structure', () => {
    expect(wrapper.exists(Context)).toBeTruthy()
    expect(wrapper.exists(BucWebSocket)).toBeTruthy()
    expect(wrapper.exists(Tag)).toBeTruthy()
    expect(wrapper.find(Tag).first().render().text()).toEqual('ui:youComeFrom' + mockPesysContext + '.')
    expect(wrapper.find(Tag).last().render().text()).toEqual('buc:form-caseType' + ': ' + mockSakType)
    expect(wrapper.exists(DivWithLinks)).toBeTruthy()
    expect(wrapper.find(DivWithLinks).render().text()).toEqual('ui:lawsource' + 'â€¢' + 'ui:help')
  })
})
