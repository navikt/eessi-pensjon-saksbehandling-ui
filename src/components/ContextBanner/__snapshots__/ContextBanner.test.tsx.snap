// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ContextBanner/ContextBanner Render: match snapshot 1`] = `
.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: inherit;
  line-height: inherit;
  color: #0067c5 !important;
}

.c11:focus {
  background-color: #19548a !important;
  color: #FFFFFF !important;
}

.c11 svg {
  fill: #0067c5 !important;
  stroke: #0067c5 !important;
}

.c11 svg path {
  stroke: #0067c5 !important;
}

.c11 svg:focus {
  fill: #FFFFFF !important;
  stroke: #FFFFFF !important;
}

ui-fss.g16[id="c10"] {
  content: "c11,";
}

.c7 {
  display: inline-block;
  margin-left: 0.25rem;
}

.c12 {
  display: inline-block;
  margin-left: 0.5rem;
}

ui-fss.g25[id="c6"] {
  content: "c7,c12,";
}

.c9 {
  padding: 0.5rem 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

ui-fss.g31[id="c8"] {
  content: "c9,";
}

.c14 {
  padding: 0rem 0.5rem;
}

ui-fss.g32[id="c13"] {
  content: "c14,";
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  background-color: white;
}

ui-fss.g33[id="c0"] {
  content: "c1,";
}

.c3 {
  padding: 0.5rem 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 * {
  font-size: inherit;
  line-height: inherit;
}

ui-fss.g34[id="c2"] {
  content: "c3,";
}

.c5 {
  padding: 0rem 0.5rem;
  margin-left: 0.25rem;
  margin-right: 0.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

ui-fss.g35[id="c4"] {
  content: "c5,";
}

<ContextBanner
  highContrast={false}
  mode="buclist"
>
  <q
    highContrast={false}
  >
    <Component
      theme={
        Object {
          "alternative-background-color": "#FFFFFF",
          "alternative-hover-color": "#cce1f3",
          "black": "#000000",
          "border-radius-base": "4px",
          "border-radius-large": "6px",
          "border-radius-small": "2px",
          "container-desktop": "(940px + @grid-gutter-width)",
          "container-large-desktop": "(1140px + @grid-gutter-width)",
          "container-lg": "(1140px + @grid-gutter-width)",
          "container-md": "(940px + @grid-gutter-width)",
          "container-sm": "(720px + @grid-gutter-width)",
          "container-tablet": "(720px + @grid-gutter-width)",
          "fokusFarge": "#19548a",
          "grayBackground": "#e9e7e7",
          "grayIcon": "#b7b1a9",
          "grayInactive": "#78706a",
          "grayModia": "#333333",
          "grid-columns": "12",
          "grid-float-breakpoint": "768px",
          "grid-float-breakpoint-max": "(@grid-float-breakpoint - 1)",
          "grid-gutter-width": "16px",
          "inverted-font-color": "#FFFFFF",
          "line-height-large": "1.3333333",
          "line-height-small": "1.5",
          "main-active-color": "#005b82",
          "main-background-color": "#f1f1f1",
          "main-border-color": "#78706a",
          "main-border-radius": "4px",
          "main-border-width": "1px",
          "main-disabled-color": "#b7b1a9",
          "main-error-color": "#ba3a26",
          "main-focus-color": "#19548a",
          "main-font-color": "#3e3832",
          "main-hover-color": "#0074df",
          "main-interactive-color": "#0067c5",
          "main-success-color": "#06893a",
          "main-warning-color": "#ff9100",
          "navBakgrunn": "#ffffff",
          "navBla": "#0067c5",
          "navBlaDarken20": "#32414f",
          "navBlaDarken40": "#254b6d",
          "navBlaDarken60": "#19548a",
          "navBlaDarken80": "#0c5ea8",
          "navBlaLighten20": "#cce1f3",
          "navBlaLighten40": "#99c2e8",
          "navBlaLighten60": "#66a4dc",
          "navBlaLighten80": "#3385d1",
          "navDypBla": "#005b82",
          "navDypBlaDarken20": "#323f42",
          "navDypBlaDarken40": "#254652",
          "navDypBlaDarken60": "#194d62",
          "navDypBlaDarken80": "#0c5472",
          "navDypBlaLighten20": "#ccdee6",
          "navDypBlaLighten40": "#99bdcd",
          "navDypBlaLighten60": "#669db4",
          "navDypBlaLighten80": "#337c9b",
          "navGra20": "#c6c2bf",
          "navGra40": "#b7b1a9",
          "navGra60": "#78706a",
          "navGra80": "#59514b",
          "navGraBakgrunn": "#f1f1f1",
          "navGronn": "#06893a",
          "navGronnDarken20": "#334834",
          "navGronnDarken40": "#285835",
          "navGronnDarken60": "#1c6937",
          "navGronnDarken80": "#117938",
          "navGronnLighten20": "#cde7d8",
          "navGronnLighten40": "#9bd0b0",
          "navGronnLighten60": "#6ab889",
          "navGronnLighten80": "#38a161",
          "navLilla": "#634689",
          "navLillaDarken20": "#453b43",
          "navLillaDarken40": "#4d3e55",
          "navLillaDarken60": "#544066",
          "navLillaDarken80": "#5c4378",
          "navLillaLighten20": "#e0dae7",
          "navLillaLighten40": "#c1b5d0",
          "navLillaLighten60": "#a190b8",
          "navLillaLighten80": "#826ba1",
          "navLimeGronn": "#a2ad00",
          "navLimeGronnDarken20": "#524f28",
          "navLimeGronnDarken40": "#66671e",
          "navLimeGronnDarken60": "#7a7e14",
          "navLimeGronnDarken80": "#8e960a",
          "navLimeGronnLighten20": "#ecefcc",
          "navLimeGronnLighten40": "#dade99",
          "navLimeGronnLighten60": "#c7ce66",
          "navLimeGronnLighten80": "#b5bd33",
          "navLysBla": "#66cbec",
          "navLysBlaDarken20": "#465557",
          "navLysBlaDarken40": "#4e737c",
          "navLysBlaDarken60": "#5690a2",
          "navLysBlaDarken80": "#5eaec7",
          "navLysBlaLighten20": "#e0f5fb",
          "navLysBlaLighten40": "#c2eaf7",
          "navLysBlaLighten60": "#a3e0f4",
          "navLysBlaLighten80": "#85d5f0",
          "navLysGra": "#e9e7e7",
          "navMorkGra": "#3e3832",
          "navOransje": "#ff9100",
          "navOransjeDarken20": "#654a28",
          "navOransjeDarken40": "#8b5c1e",
          "navOransjeDarken60": "#b26d14",
          "navOransjeDarken80": "#d87f0a",
          "navOransjeLighten20": "#ffe9cc",
          "navOransjeLighten40": "#ffd399",
          "navOransjeLighten60": "#ffbd66",
          "navOransjeLighten80": "#ffa733",
          "navRod": "#c30000",
          "navRodDarken20": "#592d28",
          "navRodDarken40": "#73221e",
          "navRodDarken60": "#8e1614",
          "navRodDarken80": "#a80b0a",
          "navRodLighten20": "#f3cccc",
          "navRodLighten40": "#e79999",
          "navRodLighten60": "#db6666",
          "navRodLighten80": "#cf3333",
          "orangeFocus": "#ffbd66",
          "orangeFocusDarken20": "#65533c",
          "orangeFocusDarken40": "#8b6d47",
          "orangeFocusDarken60": "#b28851",
          "orangeFocusDarken80": "#d8a25c",
          "orangeFocusLighten20": "#fff2e0",
          "orangeFocusLighten40": "#ffe5c2",
          "orangeFocusLighten60": "#ffd7a3",
          "orangeFocusLighten80": "#ffca85",
          "padding-base-horizontal": "12px",
          "padding-base-vertical": "6px",
          "padding-large-horizontal": "16px",
          "padding-large-vertical": "10px",
          "padding-small-horizontal": "10px",
          "padding-small-vertical": "5px",
          "padding-xs-horizontal": "5px",
          "padding-xs-vertical": "1px",
          "pinkErrorBg": "#f3e3e3",
          "redError": "#ba3a26",
          "redErrorDarken20": "#573830",
          "redErrorDarken40": "#70392d",
          "redErrorDarken60": "#88392b",
          "redErrorDarken80": "#a13a28",
          "redErrorLighten20": "#f1d8d4",
          "redErrorLighten40": "#e3b0a8",
          "redErrorLighten60": "#d6897d",
          "redErrorLighten80": "#c86151",
          "screen-desktop": "992px",
          "screen-lg": "1200px",
          "screen-lg-desktop": "1200px",
          "screen-lg-min": "1200px",
          "screen-md": "992px",
          "screen-md-max": "(@screen-lg-min - 1)",
          "screen-md-min": "992px",
          "screen-sm": "768px",
          "screen-sm-max": "(@screen-md-min - 1)",
          "screen-sm-min": "768px",
          "screen-tablet": "768px",
          "screen-xs-max": "(@screen-sm-min - 1)",
          "type": "theme",
          "white": "#FFFFFF",
        }
      }
    >
      <styled.div>
        <div
          className="c0 c1"
        >
          <styled.div>
            <div
              className="c2 c3"
            >
              <Component
                avdodFnr=""
              >
                <div
                  className="websocket"
                />
              </Component>
              <HoyreChevron>
                <NavFrontendChevron
                  className=""
                  stor={false}
                  type="hÃ¸yre"
                >
                  <span
                    className="nav-frontend-chevron chevronboks chevron--hoyre"
                  />
                </NavFrontendChevron>
              </HoyreChevron>
              <Styled(Element)>
                <Element
                  className="c4 c5"
                >
                  <TypografiBase
                    className="c4 c5"
                    type="element"
                  >
                    <p
                      className="typo-element c4 c5"
                    >
                      <span>
                        ui:youComeFrom
                      </span>
                      <styled.div
                        data-size="0.25"
                      >
                        <div
                          className="c6 c7"
                          data-size="0.25"
                        />
                      </styled.div>
                      <strong>
                        vedtakskontekst
                      </strong>
                      .
                    </p>
                  </TypografiBase>
                </Element>
              </Styled(Element)>
              <Styled(Element)>
                <Element
                  className="c4 c5"
                >
                  <TypografiBase
                    className="c4 c5"
                    type="element"
                  >
                    <p
                      className="typo-element c4 c5"
                    >
                      <span>
                        buc:form-caseType
                        : 
                      </span>
                      <styled.div
                        data-size="0.25"
                      >
                        <div
                          className="c6 c7"
                          data-size="0.25"
                        />
                      </styled.div>
                      <strong>
                        Generell
                      </strong>
                    </p>
                  </TypografiBase>
                </Element>
              </Styled(Element)>
            </div>
          </styled.div>
          <styled.div>
            <div
              className="c8 c9"
            >
              <Styled(s)
                data-amplitude="links.rettskilder"
                href="https://lovdata.no/pro/#document/NAV/rundskriv/v2-45-03"
                onClick={[Function]}
                target="_blank"
              >
                <s
                  className="c10 c11"
                  data-amplitude="links.rettskilder"
                  href="https://lovdata.no/pro/#document/NAV/rundskriv/v2-45-03"
                  onClick={[Function]}
                  target="_blank"
                >
                  <a
                    className="lenke c10 c11"
                    data-amplitude="links.rettskilder"
                    href="https://lovdata.no/pro/#document/NAV/rundskriv/v2-45-03"
                    onClick={[Function]}
                    rel="noopener"
                    target="_blank"
                  >
                    ui:lawsource
                    <styled.div
                      data-size="0.5"
                    >
                      <div
                        className="c6 c12"
                        data-size="0.5"
                      />
                    </styled.div>
                    <LogoutIcon
                      color="#0067c5"
                    >
                      <svg
                        color="#0067c5"
                        contentScriptType="text/ecmascript"
                        contentStyleType="text/css"
                        enableBackground="new 0 0 24 24"
                        height="24px"
                        id="Outline_Version"
                        preserveAspectRatio="xMidYMid meet"
                        version="1.1"
                        viewBox="0 0 24 24"
                        width="24px"
                        x="0px"
                        xmlns="http://www.w3.org/2000/svg"
                        y="0px"
                        zoomAndPan="magnify"
                      >
                        <g
                          fill="#0067c5"
                        >
                          <path
                            d="M13.5,18c-0.276,0-0.5,0.224-0.5,0.5V22H1V2h12v3.5C13,5.776,13.224,6,13.5,6S14,5.776,14,5.5v-4C14,1.224,13.776,1,13.5,1   h-13C0.224,1,0,1.224,0,1.5v21C0,22.776,0.224,23,0.5,23h13c0.276,0,0.5-0.224,0.5-0.5v-4C14,18.224,13.776,18,13.5,18z"
                          />
                          <path
                            d="M23.961,12.691c0.051-0.122,0.051-0.26,0-0.382c-0.025-0.062-0.062-0.117-0.108-0.163l-5-5 c-0.195-0.195-0.512-0.195-0.707,0s-0.195,0.512,0,0.707L22.293,12H5.5C5.224,12,5,12.224,5,12.5S5.224,13,5.5,13h16.793   l-4.146,4.146c-0.195,0.195-0.195,0.512,0,0.707C18.244,17.951,18.372,18,18.5,18s0.256-0.049,0.354-0.146l5-5   C23.899,12.808,23.936,12.752,23.961,12.691z"
                          />
                        </g>
                      </svg>
                    </LogoutIcon>
                  </a>
                </s>
              </Styled(s)>
              <styled.span>
                <span
                  className="c13 c14"
                >
                  â¢
                </span>
              </styled.span>
              <Styled(s)
                data-amplitude="links.hjelpe"
                href="https://navno.sharepoint.com/sites/fag-og-ytelser-regelverk-og-rutiner/SitePages/Pensjon-.aspx"
                onClick={[Function]}
                target="_blank"
              >
                <s
                  className="c10 c11"
                  data-amplitude="links.hjelpe"
                  href="https://navno.sharepoint.com/sites/fag-og-ytelser-regelverk-og-rutiner/SitePages/Pensjon-.aspx"
                  onClick={[Function]}
                  target="_blank"
                >
                  <a
                    className="lenke c10 c11"
                    data-amplitude="links.hjelpe"
                    href="https://navno.sharepoint.com/sites/fag-og-ytelser-regelverk-og-rutiner/SitePages/Pensjon-.aspx"
                    onClick={[Function]}
                    rel="noopener"
                    target="_blank"
                  >
                    ui:help
                    <styled.div
                      data-size="0.5"
                    >
                      <div
                        className="c6 c12"
                        data-size="0.5"
                      />
                    </styled.div>
                    <LogoutIcon
                      color="#0067c5"
                    >
                      <svg
                        color="#0067c5"
                        contentScriptType="text/ecmascript"
                        contentStyleType="text/css"
                        enableBackground="new 0 0 24 24"
                        height="24px"
                        id="Outline_Version"
                        preserveAspectRatio="xMidYMid meet"
                        version="1.1"
                        viewBox="0 0 24 24"
                        width="24px"
                        x="0px"
                        xmlns="http://www.w3.org/2000/svg"
                        y="0px"
                        zoomAndPan="magnify"
                      >
                        <g
                          fill="#0067c5"
                        >
                          <path
                            d="M13.5,18c-0.276,0-0.5,0.224-0.5,0.5V22H1V2h12v3.5C13,5.776,13.224,6,13.5,6S14,5.776,14,5.5v-4C14,1.224,13.776,1,13.5,1   h-13C0.224,1,0,1.224,0,1.5v21C0,22.776,0.224,23,0.5,23h13c0.276,0,0.5-0.224,0.5-0.5v-4C14,18.224,13.776,18,13.5,18z"
                          />
                          <path
                            d="M23.961,12.691c0.051-0.122,0.051-0.26,0-0.382c-0.025-0.062-0.062-0.117-0.108-0.163l-5-5 c-0.195-0.195-0.512-0.195-0.707,0s-0.195,0.512,0,0.707L22.293,12H5.5C5.224,12,5,12.224,5,12.5S5.224,13,5.5,13h16.793   l-4.146,4.146c-0.195,0.195-0.195,0.512,0,0.707C18.244,17.951,18.372,18,18.5,18s0.256-0.049,0.354-0.146l5-5   C23.899,12.808,23.936,12.752,23.961,12.691z"
                          />
                        </g>
                      </svg>
                    </LogoutIcon>
                  </a>
                </s>
              </Styled(s)>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </Component>
  </q>
</ContextBanner>
`;
